linker/linker_test.go

```diff
@@ -1067,6 +1067,30 @@ func TestLinkerValidation(t *testing.T) {
 			},
 			expectedErr: "foo.proto:9:6: message foo.bar.Baz: option (foo.bar.any): could not resolve type reference type.googleapis.com/Foo",
 		},
+		"failure_any_message_literal_duplicate": {
+			input: map[string]string{
+				"foo.proto": `
+					syntax = "proto3";
+					package foo.bar;
+					import "google/protobuf/any.proto";
+					import "google/protobuf/descriptor.proto";
+					message Foo { string a = 1; int32 b = 2; }
+					extend google.protobuf.MessageOptions { optional google.protobuf.Any any = 10001; }
+					message Baz {
+					  option (any) = {
+					    [type.googleapis.com/foo.bar.Foo] <
+					      a: "abc"
+					      b: 123
+					    >
+					    [type.googleapis.com/foo.bar.Foo] <
+					      a: "abc"
+					      b: 123
+					    >
+					  };
+					}`,
+			},
+			expectedErr: "foo.proto:13:6: message foo.bar.Baz: option (foo.bar.any): multiple any type references are not allowed",
```

Comment:
It felt better to provide an error at the location of the duplicate Any option instead of the first one.